guides/source/active_storage_overview.md

```diff
@@ -641,6 +638,45 @@ and then generate routes like this:
 <%= cdn_image_url(user.avatar.variant(resize_to_limit: [128, 128])) %>
 ```
 
+### Authenticated Controllers
+
+The default Active Storage controllers are publicly accessible. The generated
+URLs use a plain [`signed_id`][ActiveStorage::Blob#signed_id], making them hard to
+guess but permanent. Anyone that knows the blob URL will be able to access it,
+even if a `before_action` in your `ApplicationController` would otherwise
+require a login. If your files require a higher level of protection consider
+implementing your own
+[`ActiveStorage::Blobs::RedirectController`](ActiveStorage::Blobs::RedirectController) and
+[`ActiveStorage::Representations::RedirectController`](ActiveStorage::Representations::RedirectController)
+
+For example:
+
+```ruby
```

Comment:
This example is useful too, but I'm not sure it's what the guides suggest. I was expecting something like this:

```ruby
class MyRedirectController < ActiveStorage::Representations::RedirectController
  def show
     super if Current.user.can_view?(@blob)
  end
end
```

Not opposed to the example here, it is the Basecamp one too, but I think if you go with this you shouldn't describe this as "implementing your own controller" since those controllers aren't referenced anywhere in the example.

Reply:
Yes, good catch.
Inheriting and using `super` will still use the Cache-Control header, which might not be what you'd expect when you want extra security:
https://github.com/rails/rails/blob/main/activestorage/app/controllers/active_storage/blobs/redirect_controller.rb#L11

I think it makes more sense to leave this decision to the developer instead of recommending inheritance (which doesn't add much in this case). I'll try to change the wording a bit.

Reply:
The original PR for the warning is here: https://github.com/rails/rails/pull/40675
Another related issue: https://github.com/rails/rails/pull/41505

Reply:
I've changed the wording a bit.